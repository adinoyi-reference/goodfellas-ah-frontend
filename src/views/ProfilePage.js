import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Body from '../components/layout/DefaultLayout';
import Loading from '../components/shared/Loading';
import ProfileToolbar from '../containers/ProfileToolbar';
import FollowerList from '../components/profile/FollowerList';
import FollowingList from '../components/profile/FollowingList';
import ProfileArticleList from '../components/article/ArticlesByAuthor';
import ProfileFavoriteList from '../components/article/AuthorFavorites';
import { fetchProfile } from '../actions/profileActions';
import { userPlaceholderImage } from '../mixin';

class Profile extends Component {
  componentDidMount() {
    const { fetchProfile: profileFetch } = this.props;
    profileFetch(1);
  }

  render() {
    const { profileStore } = this.props;
    if (profileStore.loading) return <Loading />;
    if (profileStore.profileError) return <div>{profileStore.profileError}</div>;
    const fullName = `${profileStore.user.firstname} ${profileStore.user.lastname}`;
    return (
      <Body className="profile">
        <div>
          <header>
            <img alt="profile" className="profile-image" src={profileStore.profile.image || userPlaceholderImage} />
            <h3 id="user-name" className="username">{fullName}</h3>
            <Link className="button outline" to="/profile/edit">Edit Profile</Link>
          </header>
          <ProfileToolbar profile={profileStore} />
          {(() => {
            switch (profileStore.profileView) {
              case 'Followers':
                return <FollowerList userFullName={fullName} followers={profileStore.followers} />;
              case 'Articles':
                return (
                  <ProfileArticleList
                    author={fullName}
                    authorImage={profileStore.profile.image}
                    articles={profileStore.articles}
                    userFullName={fullName}
                  />
                );
              case 'Favorites':
                return (
                  <ProfileFavoriteList
                    author={fullName}
                    authorImage={profileStore.profile.image}
                    articles={profileStore.favorites}
                    userFullName={fullName}
                  />);
              case 'Following':
              default:
                return <FollowingList userFullName={fullName} following={profileStore.following} />;
            }
          })()
          }
        </div>
      </Body>
    );
  }
}
const mapStateToProps = (state) => ({
  profileStore: state.profile,
});

export default connect(
  mapStateToProps,
  { fetchProfile },
)(Profile);
