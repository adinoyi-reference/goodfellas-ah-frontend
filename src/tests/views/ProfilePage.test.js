import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfilePage from '../../views/ProfilePage';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <ProfilePage />
    </Root>,
  );
});

afterEach(() => wrapped.unmount());

describe('Profile UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
