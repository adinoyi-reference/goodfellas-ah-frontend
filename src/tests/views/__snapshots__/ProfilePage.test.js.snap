// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile UI render features view should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Profile)>
      <Profile
        fetchProfile={[Function]}
        profileStore={
          Object {
            "articles": Array [],
            "favorites": Array [],
            "followers": Array [],
            "following": Array [],
            "loading": true,
            "profile": Object {},
            "profileError": "",
            "profileView": "Following",
            "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MywiaWF0IjoxNTQxNDA3NzcxLCJleHAiOjE1NDE0OTQxNzF9.DHaysbXDOqw8iWXw_JKE8-9H-s2AXS_dqAz1NzO-rpg",
            "user": Object {},
            "userId": 1,
          }
        }
      >
        <_default>
          <div
            className="loading"
          >
            Loading.......
          </div>
        </_default>
      </Profile>
    </Connect(Profile)>
  </Provider>
</_default>
`;
